R/eda.R

```diff
@@ -28,20 +28,73 @@ generate_report <- function(dataframe, cat_vars, num_vars) {
 #'
 #' @param dataframe tbl. The dataframe to be inspected.
 #' @param num_vars vector of character strings of the names of the numeric variables.
-#' @param plot logical. If TRUE (the default) the distribution of the numeric variables will be plotted.
 #'
-#' @return tbl. A dataframe contains the statistical summary.
+#' @return list. A list of a dataframe contains the statistical summary and the ggplot objects of the faceted histogram of numeric variables.
 #'
 #' @export
 #'
 #' @examples
-#' X <- tibble(type = c('Car','Bus','Car'), height = c(10,20,30))
-#' num_vars <- c('height')
-#' describe_num_var(X, num_vars)
-#'
-describe_num_var <- function(dataframe, num_vars, plot=TRUE) {
-  #TODO implement function
-  print(num_vars)
+#' X <- tibble(type = c('Car', 'Bus', 'Car'), height = c(10, 20, 15), width = c(10, 15, 13))
+#' num_vars <- c('height', 'width')
+#' result <- describe_num_var(X, num_vars)
+#' result$summary
+#' result$plot
+#'
+describe_num_var <- function(dataframe, num_vars) {
+  # Check the input dataframe is a dataframe
+  if (!is.data.frame(dataframe)) {
```

Comment:
Will this check throw error for tibble also? If it throws error for tibble, then you can modify it to accept tibble also.

Reply:
No, tibble is dataframe, so it won't throw an error for tibble.

Reply:
I tried it on a dataframe and a tibble and it worked for me.